appinventor/components/src/com/google/appinventor/components/runtime/FirebaseDB.java

```diff
@@ -0,0 +1,347 @@
+  // stipulated that the property should appear with a default value.  Note
+  // that this default specifies what should be shown in the designer:
+  // it does not automatically set the value of FirebaseURL by itself,
+  // which is why we explicitly set the variable FirebaseURL above
+  // where the component is created.
+
+  /**
+   * Returns the URL of the firebase.
+   */
+  @SimpleProperty(
+      category = PropertyCategory.BEHAVIOR)
+  public String FirebaseURL() {
+    return firebaseURL;
+  }
+
+  /**
+   * Specifies the URL of the firebase.
+   * The default value is currently my private firebase url //TODO: this should be changed
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_STRING,
+      defaultValue = "https://incandescent-fire-4621.firebaseio.com/")
+  @SimpleProperty
+  public void FirebaseURL(String url) {
+    firebaseURL = url;
+    resetListener();
+  }
+  
+  
+  // The two procedures below give the getter and setter for the
+  // Firebase component's ProjectPath property.  Each one has
+  // a @SimpleProperty annotation to indicate that it's a property in
+  // the language (and blocks will be generated for it).  The setter
+  // also has a @DesignerProperty that makes this property appear in the
+  // Properties listed with the component in the designer.  Here we've
+  // stipulated that the property should appear with a default value.  Note
+  // that this default specifies what should be shown in the designer:
+  // it does not automatically set the value of ProjectPath by itself,
+  // which is why we explicitly set the variable ProjectPath above
+  // where the component is created.
+  /**
+   * Returns the path of the firebase.
+   */
+  @SimpleProperty(
+      category = PropertyCategory.BEHAVIOR)
+  public String ProjectPath() {
+    return projectPath;
+  }
+
+  /**
+   * Specifies the path of the firebase.
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_STRING,
+      defaultValue = "")
+  @SimpleProperty
+  public void ProjectPath(String path) {
+    projectPath = path;    
+    resetListener();
+  }
+  
+  private void resetListener() {
+    // remove listeners from the old firebase path
+    myFirebase.removeEventListener(listener);
+    // set listeners for the new firebase path
+    myFirebase = new Firebase(firebaseURL+projectPath);
+    myFirebase.addChildEventListener(listener);
+  }
+
+  /**
+   * Asks Firebase to store the given value under the given tag
+   *
+   * @param tag The tag to use
+   * @param valueToStore The value to store. Can be any type of value (e.g.
+   * number, text, boolean or list).
+   */
+  @SimpleFunction
+  // The @SimpleFunction annotation arranges for this to be a
+  // function (StoreValue)  associated with the component.
+  public void StoreValue(final String tag, final Object valueToStore) {
+    this.myFirebase.child(tag).setValue(valueToStore);
```

Comment:
What happens if the write fails? 
I had a quick look at the API and it seems that it does not use Exceptions (which makes sense in an async API) but this method is not using a callback, which seems to be the way the API returns errors, so the user will never know if the value has been saved. There is a setValue(value, callback) version of the method which makes more sense to use in here, even if it's not exposed to AI devs though blocks, but at least to do something with the error if it happens.